### PR TITLE
[D2IQ-72015] Fix storage race condition for Kommander addon

### PR DESCRIPTION
**Pull request by shaneutt**
**What type of PR is this?**
Bug

**What this PR does/ why we need it**:
- bug: fix broken annotation for Kommander addon
- bug: fix storage race condition for Kommander addon

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-72015

**Does this PR introduce a user-facing change?**:
```release-note
- the kommander addon will now wait for a default storage class before deploying in order to support the storage needed for kubecost components
```

**Checklist**

- [x] The commit message explains the changes and why are needed.
- [x] The code builds and passes lint/style checks locally.
- [x] The relevant subset of integration tests pass locally.
- [x] The core changes are covered by tests.
- [x] The documentation is updated where needed.

### CODE REVIEW

**Review by makkes — APPROVED** [review]

**Comment by jimmidyson** [comment]
Sorry what was broken about the upgrade strategy annotation? This seemed to be working fine, even though I agree it looks better the way you've got it now... (and not discussing why we need delete strategy for upgrade which is breaking upgrades, but anyway...).

File: addons/kommander/1.2/kommander.yaml [hunk 1]
```diff
@@ -19,6 +19,10 @@ metadata:
     docs.kubeaddons.mesosphere.io/kommander-grafana: "https://grafana.com/docs/"
     values.chart.helm.kubeaddons.mesosphere.io/kommander: "https://raw.githubusercontent.com/mesosphere/charts/d71f1e0/stable/kommander/values.yaml"
     helmv2.kubeaddons.mesosphere.io/upgrade-strategy: '[{"upgradeFrom": "<=0.1.22", "strategy": "delete"}]'
+    # TODO: we're temporarily supporting dependency on an existing default storage class
+    # on the cluster, this hack will trigger re-queue on Addons until one exists.
+    # This was originally added to support the PVC's needed for the Kubecost subcomponent.
+    kubeaddons.mesosphere.io/hack-requires-defaultstorageclass: "true"
```
Comment (jimmidyson) [inline addons/kommander/1.2/kommander.yaml:25]: What part of this addon requires default storage class?

Comment (gracedo) [reply addons/kommander/1.2/kommander.yaml:25]: Kubecost components

Comment (shaneutt) [reply addons/kommander/1.2/kommander.yaml:25]: :point_up: yep, kubecost. It's in the comment above the annotation.

**Comment by shaneutt** [comment]
> Sorry what was broken about the upgrade strategy annotation? This seemed to be working fine, even though I agree it looks better the way you've got it now... (and not discussing why we need delete strategy for upgrade which is breaking upgrades, but anyway...).

Nope, it was JUST to make it look better :+1:

File: addons/kommander/1.2/kommander.yaml [hunk 2]
```diff
@@ -18,8 +18,11 @@ metadata:
     docs.kubeaddons.mesosphere.io/karma: "https://github.com/prymitive/karma"
     docs.kubeaddons.mesosphere.io/kommander-grafana: "https://grafana.com/docs/"
     values.chart.helm.kubeaddons.mesosphere.io/kommander: "https://raw.githubusercontent.com/mesosphere/charts/d71f1e0/stable/kommander/values.yaml"
-    helmv2.kubeaddons.mesosphere.io/upgrade-strategy: '[{"upgradeFrom": "<=0.1.22",
```
Comment (gracedo) [inline addons/kommander/1.2/kommander.yaml:21]: this line break was added by the automated chart bump job, if we would rather it not be adding line breaks maybe we can ask @d2iq-dispatch to fix the script

**Review by gracedo — APPROVED** [review]

**Review by hectorj2f — APPROVED** [review]

**Pull request merged**